src/shop-redux-store.js

```diff
@@ -0,0 +1,301 @@
+import createStore from '../node_modules/@0xcda7a/redux-es6/es/createStore.js';
+
+const store = createStore(
+  (state, action) => {
+    let result = state;
+
+    switch (action.type) {
+      // window.location changed
+      case '_pathChanged':
+        {
+          const pathParts = action.path.slice(1).split('/');
+          const page = pathParts[0];
+          const categoryName = pathParts[1];
+          const itemName = pathParts[2];
+          const category = findCategory(state.categories, categoryName);
+          loadCategory(category);
```

Comment:
Side-effects would ideally be encapsulated in middleware or action creators. Reducers should be synchronous and free of side-effects.